geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -127,6 +128,50 @@ private static MemberSet calculateDiff(RegionProvider regionProvider, List<Redis
     return diff;
   }
 
+  public static Set<byte[]> sinter(RegionProvider regionProvider, List<RedisKey> keys) {
+    List<RedisSet> sets = createRedisSetList(keys, regionProvider);
+    RedisSet smallestSet = findSmallest(sets);
+    if (smallestSet.scard() == 0) {
+      return Collections.emptySet();
+    }
+
+    MemberSet result = new MemberSet(smallestSet.scard());
+    for (byte[] member : smallestSet.members) {
+      boolean addToSet = true;
+      for (RedisSet otherSet : sets)
+        if (!otherSet.equals(smallestSet) && !otherSet.members.contains(member)) {
```

Comment:
don't use "equals()" here. It is too expensive. I think `if (otherSet == smallestSet) {continue;}` is what you want. This is just an optimization (the code would work without this check) but it is waste of time. But by using "equals()" we even waste more time.